tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -99,6 +106,38 @@
 };
 """)
 
+METHOD_DEFINITION = CodeTemplate("""\
+${return_type} ${type_wrapper_name}(${formals}) {
+  ${type_definition_body}
+}
+""")
+
+WRAPPER_REGISTRATION = CodeTemplate("""\
+m.impl("${unqual_operator_name_with_overload}",
+       TORCH_FN(${class_type}::${type_wrapper_name})
+);
+""")
+
+INPLACE_REDISPATCH = CodeTemplate("""\
+{
+  at::AutoDispatchBelowInplaceOrView guard(true);
```

Comment:
Even if you decide not to actually remove the ctor param, I'd suggest we not supply it explicitly at callsites, again to clarify what the guard actually does.